src/st_stuff.c

```diff
@@ -615,65 +615,111 @@ static void ST_drawDebugInfo(void)
 
 static void ST_drawScore(void)
 {
-	// SCORE:
-	ST_DrawPatchFromHud(HUD_SCORE, sboscore);
+	const INT32 v_splitflag = (splitscreen && stplyr == &players[displayplayer] ? V_SPLITSCREEN : 0);
+	const INT32 v_splitoffset = splitscreen ? (v_splitflag ? 0 : BASEVIDHEIGHT/2) : 0;
+
+	INT32 scoreflags = hudinfo[HUD_SCORE].flags|v_splitflag;
+	if (splitscreen && stplyr == &players[secondarydisplayplayer])
+		scoreflags &= ~(V_SNAPTOTOP);
+
+	V_DrawScaledPatch(hudinfo[HUD_SCORE].x, hudinfo[HUD_SCORE].y + v_splitoffset, scoreflags, sboscore);
+
+	//ST_DrawPatchFromHud(HUD_SCORE, sboscore);
 	if (objectplacing)
 	{
 		if (op_displayflags > UINT16_MAX)
-			ST_DrawOverlayPatch(SCX(hudinfo[HUD_SCORENUM].x-tallminus->width), SCY(hudinfo[HUD_SCORENUM].y), tallminus);
+			V_DrawFixedPatch((hudinfo[HUD_SCORENUM].x-tallminus->width) * FRACUNIT, (hudinfo[HUD_SCORENUM].y) * FRACUNIT, FRACUNIT, hudinfo[HUD_SCORENUM].flags, tallminus, (void *) 0);
+			//ST_DrawOverlayPatch(SCX(hudinfo[HUD_SCORENUM].x-tallminus->width), SCY(hudinfo[HUD_SCORENUM].y), tallminus);
 		else
-			ST_DrawNumFromHud(HUD_SCORENUM, op_displayflags);
+			V_DrawTallNum(hudinfo[HUD_SCORENUM].x, hudinfo[HUD_SCORENUM].y, hudinfo[HUD_SCORENUM].flags, op_displayflags);
+			//ST_DrawNumFromHud(HUD_SCORENUM, op_displayflags);
```

Comment:
Consider removing comments like these so the code is less cluttered - if we want to go back to the old code, we could just check the git history.